### PR TITLE
registry: skip invalid Discrete values and empty Sample-s

### PR DESCRIPTION
**Pull request by mxmCherry**

### CODE REVIEW

File: registry.go [hunk 1]
```diff
@@ -149,21 +150,30 @@ func (r *Registry) Flush() error {
...
+			}
 			for _, rep := range reporters {
 				if err := rep.Discrete(name, tags, val); err != nil {
 					return err
 				}
 			}
+
 		case Sample:
 			val := inst.Snapshot()
+			if val.Count() == 0 {
```
Comment (dim) [inline registry.go:167]: That's not right. It means we will never report 0s

**Review by dim — CHANGES_REQUESTED** [review]

File: registry.go (hunk 1 above)
Comment (mxmCherry) [reply registry.go:167]: Hm, @dim I thought that 0 count distribution means "nothing happened for metric X" so it shouldn't be reported.

What about this approach then? https://github.com/bsm/instruments/pull/9

Comment (dim) [reply registry.go:167]: I think you should just remove this condition again and we are done!

**Review by dim — APPROVED** [review]

**Comment by dim** [comment]
Released as v1.3.9

**Pull request merged**